src/blockchains/filecoin.ts

```diff
@@ -0,0 +1,73 @@
+import {BlockchainHandler, BlockchainHandlerOpts} from "../blockchain-handler";
+import {AccountID} from "caip";
+import {ConsentMessage, getConsentMessage, LinkProof} from "../utils";
+import type { MessageParams } from '@zondax/filecoin-signing-tools'
+
+const namespace = 'fil'
+
+const moduleToImport = process.env.JEST_WORKER_ID ? "@zondax/filecoin-signing-tools/nodejs" : "@zondax/filecoin-signing-tools"
+
+function asTransaction(address: string, message: string): MessageParams {
+    const messageHex = Buffer.from(message).toString('hex')
+    return {
+        from: address,
+        to: address,
+        value: "0",
+        method: 0,
+        gasPrice: "1",
+        gasLimit: 1000,
+        nonce: 0,
+        params: messageHex
+    }
+}
+
+export async function createLink (did: string, account: AccountID, provider: any, opts: BlockchainHandlerOpts): Promise<LinkProof> {
+    const consentMessage = getConsentMessage(did, !opts.skipTimestamp)
+    const addresses = await provider.getAccounts()
+    const payload = asTransaction(addresses[0], JSON.stringify(consentMessage))
+    const signatureResponse = await provider.sign(account.address, payload)
+    const proof: LinkProof = {
+        version: 2,
+        type: 'eoa-tx',
+        message: consentMessage.message,
+        signature: signatureResponse.signature.data,
+        account: account.toString()
+    }
+    if (!opts.skipTimestamp) proof.timestamp = consentMessage.timestamp
+    return proof
+}
+
+export async function authenticate(message: string, account: AccountID, provider: any): Promise<string> {
+    const addresses = await provider.getAccounts()
+    const payload = asTransaction(addresses[0], JSON.stringify(message))
+    const signatureResponse = await provider.sign(account.address, payload)
+    return signatureResponse.signature.data
+}
+
+export async function validateLink (proof: LinkProof): Promise<LinkProof | null> {
+    const signingTools = await import(moduleToImport)
+    const account = new AccountID(proof.account)
+    const consentMessage: ConsentMessage = {
+        message: proof.message,
+    }
+    if (proof.timestamp) {
+        consentMessage.timestamp = proof.timestamp
+    }
+    const payload = asTransaction(account.address, JSON.stringify(consentMessage))
+    const transaction = signingTools.transactionSerialize(payload);
+    const recover = signingTools.verifySignature(proof.signature, transaction);
+    if (recover) {
+        return proof
```

Comment:
Don't we need to verify that the recovered account address is equal to proof.account?

Reply:
There is no need to do this. Payload sent to `verifySignature` is constructed based on `proof.account`: https://github.com/ceramicnetwork/js-3id-blockchain-utils/pull/24/files/d904505dab575614cb227c0d2a96dc2031bca95c#diff-d6621fec91c234edce5a1f917c9926acR56 If `proof.account` does not correspond to the account that provided signature,  `signingTools.verifySignature` would fail.

Reply:
No need to. Payload to be verified against signature is formed from `proof.account`, so if payload does not correspond to the signature, the function returns `null`. Added a test to display this.

Reply:
Ah, I see. The account address is included in the payload!